examples/tests/v3/test_matchers.py

```diff
@@ -0,0 +1,48 @@
+from pathlib import Path
+
+from examples.tests.v3.basic_flask_server import start_provider
+from pact.v3 import Pact, Verifier, matchers
```

Comment:
I would want to avoid polluting the namespace, so I agree with your choice of importing a module and calling `matchers.like` and `matchers.regex` etc.

My nit here is that I would want to name it `match` so that the syntax is `match.regex(...)` and `match.like(...)`, etc. This way, it reads a bit more like English :)

Reply:
I feel like the module that defines the actual classes (Matcher, MatcherEncoder, etc) should be called "matchers" since they are objects. I'd be happy to create a module called match and move the functions that return those matchers into that module.

Reply:
Just started going through your updated PR. I'm still thinking that we should expose the the functions behind a common `match.{func}` interface, as I do think it becomes most intuitive. It also serves as a nice way to distinguish `match.{func}` from similar `generate.{func}`.

I also agree that having `matchers` as the module name for the Matcher classes makes sense. Having said that, I don't think people would often (or ever?) need to interact with the class directly. So I'm liking what you've done whereby the `matchers` are defined in a submodule, and functions re-exported at a higher level.